src/spark/utils.py

```diff
@@ -142,6 +146,11 @@ def get_spark_session(
         spark_conf.set(key, value)
     spark = SparkSession.builder.config(conf=spark_conf).getOrCreate()
 
+    if scheduler_pool not in SPARK_POOLS:
+        print(f"Warning: Scheduler pool {scheduler_pool} is not in the list of available pools: {SPARK_POOLS} "
+              f"Defaulting to 'default' pool")
+    spark.sparkContext.setLocalProperty("spark.scheduler.pool", scheduler_pool)
```

Comment:
```suggestion
    if scheduler_pool not in SPARK_POOLS:
        print(f"Warning: Scheduler pool {scheduler_pool} is not in the list of available pools: {SPARK_POOLS} "
              f"Defaulting to 'default' pool")
        scheduler_pool = "default"  # might want to make this a constant
    spark.sparkContext.setLocalProperty("spark.scheduler.pool", scheduler_pool)
```

Reply:
So if someone passes in "my_special_pool" then spark will just ignore it and use "default"? It's not clear that's what happens from the code

Reply:
👍 I don't see it's necessary personally but sure...

Reply:
Whether it's necessary or not depends on what the scheduler does when it gets an invalid pool name

Reply:
yea. Like I already said `the spark context automatically assign the pool to default if scheduler_pool is unavailable.`

Reply:
What does unavailable mean though? Null or some unrecognized string?

Reply:
I updated the code to explicitly assign `scheduler_pool`.

Reply:
I know, but you're saying that's unnecessary. I'm trying to understand why you're saying that, and if that's correct, I agree we should remove it (although leave a comment to prevent the next person having the same question as I do)

Reply:
I'm assuming here you don't care whether we leave the reassignment in or not

Reply:
👍 yea. we can leave the assignment there.